src/test_priorityq.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+INSERT_PARAMETERS = [
+    ([(7, "banana")], (6, "apple"), [(7, "banana"), (6, "apple")]),
+    ([(7, "banana")], (8, "pear"), [(8, "pear"), (7, "banana")]),
+    ([(9, "pineapple"), (7, "banana")],
+     (8, "pear"),
+     [(9, "pineapple"), (7, "banana"), (8, "pear")]),
+    ([(9, "pineapple"), (8, "pear"), (7, "banana")],
+     (8, "second pear"),
+     [(9, "pineapple"), (8, "pear"), (7, "banana"), (8, "second pear")]),
+]
+
+POP_LIST = [
+    ([(9, "pineapple"), (8, "pear"), (7, "banana"), (8, "second pear")],
+     [(9, "pineapple"), (8, "pear"), (8, "second pear"), (7, "banana")]),
+]
+
+
+def test_insert_on_empty():
+    from priorityq import PriorityQueue
+    pq = PriorityQueue()
+    pq.insert((7, "banana"))
+    assert pq._container == [(7, "banana")]
+
+
+@pytest.mark.parametrize(("populated_pq", "insert_item", "result_pq"), INSERT_PARAMETERS)
```

Comment:
Avoid long lines over 80 characters (for this class). We want to adhere to PEP8. You can continue on the next line with implicit line joining:

```python
@pytest.mark.parametrize(("populated_pq", "insert_item", "result_pq"),
                         INSERT_PARAMETERS)
```